mlflow/openai/api_request_parallel_processor.py

```diff
@@ -179,11 +187,13 @@ def num_tokens_consumed_from_request(
             )
     # more logic needed to support other API calls (e.g., edits, inserts, DALL-E)
     else:
-        raise NotImplementedError(f'API endpoint "{api_endpoint}" not implemented in this script')
+        raise NotImplementedError(f'Task "{task!s}" not implemented in this script')
 
 
 def process_api_requests(
     requests: list[dict[str, any]] = None,
+    task: OpenAIObject = openai.ChatCompletion,
+    api_token: _OAITokenHolder = None,
```

Comment:
I think this should be a required argument unless we have a `if api_token` guard.

Reply:
Good point!